### PR TITLE
Feat/sofi landing

### PR DESCRIPTION
**Pull request by TheLeoB**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts [hunk 1]
```diff
@@ -792,6 +805,9 @@ export default ({ api, coreSagas, networks }) => {
       const redirect = queryParams.get('redirect') as string
       // keeps session id consistent if logging in from mobile exchange app
       const sessionIdMobile = queryParams.get('sessionId') as string
+      const pathname = yield select(selectors.router.getPathname)
```
Comment (TheLeoB) [inline packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts:808]: check to see if i can get rid of this logic

File: packages/blockchain-wallet-v4-frontend/src/scenes/Home/Banners/SofiMigrationBanner/index.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import { Icon, Image, Text } from 'blockchain-info-components'
+import { actions, selectors } from 'data'
+import { RootState } from 'data/rootReducer'
+
+import ANNOUNCEMENTS from '../constants'
+import { BannerButton, CloseLink, Column, Copy, IconWrapper, Row, Wrapper } from '../styles'
+
+const SofiMigration = ({ buySellActions, cacheActions, fiatCurrency }: Props) => {
+  const showModal = useCallback(() => {
+    buySellActions.showModal({ origin: 'WelcomeModal' })
```
Comment (TheLeoB) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Home/Banners/SofiMigrationBanner/index.tsx:17]: remove this

File: packages/blockchain-wallet-v4-frontend/src/scenes/Home/Banners/selectors.ts [hunk 3]
```diff
@@ -242,9 +243,22 @@ export const getData = (state: RootState) => {
   let bannerToShow: BannerType = null
 
   const isKycEnabled = products?.kycVerification?.enabled
+  // SOFI
+  const sofiMigrationPending = selectors.modules.profile
```
Comment (TheLeoB) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Home/Banners/selectors.ts:247]: see if i can optimize this logic

File: packages/blockchain-wallet-v4-frontend/src/scenes/Login/Sofi/Success/index.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
+import React from 'react'
```
Comment (TheLeoB) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Login/Sofi/Success/index.tsx:1]: Delete this file

File: packages/blockchain-wallet-v4-frontend/src/scenes/VerifyEmail/index.tsx [hunk 5]
```diff
@@ -22,14 +22,16 @@ class VerifyEmailContainer extends React.PureComponent<Props> {
...
-    const { signupRedirect } = signupMetadata
+    const { isSofi, signupRedirect } = signupMetadata
     if (isEmailVerified) {
       if (hasCowboyTag) {
         // When the user has the COWBOYS_2022 tag set from the backend we want to skip
         // the user straight to the dashboard and launch them into the Cowboys promo flyout
         nextProps.routerActions.push('/home')
         nextProps.saveGoal('cowboys2022', { firstLogin: true })
         nextProps.runGoals()
+      } else if (isSofi) {
```
Comment (TheLeoB) [inline packages/blockchain-wallet-v4-frontend/src/scenes/VerifyEmail/index.tsx:33]: we may associate before email verification

File: packages/blockchain-wallet-v4-frontend/src/services/browser/index.tsx [hunk 6]
```diff
@@ -36,3 +36,13 @@ export const isBrowserSafari = () => {
     safari: '>12'
   })
 }
+
+export const isBrowserAndroid = () => {
```
Comment (TheLeoB) [inline packages/blockchain-wallet-v4-frontend/src/services/browser/index.tsx:40]: double check with android team if this is always the case for OS system

File: packages/blockchain-wallet-v4-frontend/src/services/browser/index.tsx [hunk 7]
```diff
@@ -36,3 +36,13 @@ export const isBrowserSafari = () => {
...
+  const browser = Bowser.getParser(window.navigator.userAgent)
+  if (browser.getOSName() === 'Android') return true
+}
+
+export const isBrowserIOS = () => {
```
Comment (TheLeoB) [inline packages/blockchain-wallet-v4-frontend/src/services/browser/index.tsx:45]: add a fallback to show both mobile links if anything is undefined

File: packages/blockchain-wallet-v4/src/network/api/buySell/types.ts [hunk 8]
```diff
@@ -69,6 +69,7 @@ export type BSAccountType = IBSAccountType & {
 
 export type BSBalanceType = {
   available: string
+  mainBalanceToDisplay?: string
```
Comment (TheLeoB) [inline packages/blockchain-wallet-v4/src/network/api/buySell/types.ts:72]: check why i did this

**Review by jjBlockchain — APPROVED** [review]

**Review by mrodriguez-bc — COMMENTED** [review]
Reviewed in pairing.

**Pull request merged**